### PR TITLE
Accept X or x for PR checkboxes, extra whitespaces, and be case-insensitive

### PR DESCRIPTION
**Pull request by juliogonzalez**
## What does this PR do?

Accept X or x for PR checkboxes, extra whitespaces, and be case-insensitive

### CODE REVIEW

**Comment by juliogonzalez** [comment]
Complains from gitarro are unrelated to this PR

**Comment by MalloZup** [comment]
@juliogonzalez lgtm add a whitelist to rubocop. Update it   --auto-gen-config

**Comment by juliogonzalez** [comment]
@MalloZup I'd say the complains should be researched and maybe we should add the magic comment. I guess it's complaning because of https://stackoverflow.com/a/51634367

I'd say: let' merge, let master fail, let's create a bug, and let's have a loo.

**Comment by juliogonzalez** [comment]
@srbarrios what other checkboxes?

File: lib/gitarro/backend.rb [hunk 1]
```diff
@@ -259,7 +259,7 @@ def retriggered_by_comment?(pr, context)
   # this function will check if the PR contains a checkbox
   # for retrigger all the tests.
   def retriggered_by_checkbox?(pr, context)
-    return false unless pr.body.include? "[x] Re-run test \"#{context}\""
+    return false unless pr.body.match(/\[x\]\s+Re-run\s+test\s+#{context}/i)
```
Comment (juliogonzalez) [inline lib/gitarro/backend.rb]: @srbarrios Do we have checkboxes other than for the changelog ignore, and for the tests?

Comment (srbarrios) [reply lib/gitarro/backend.rb]: Nope. We don't. Right :) Only comments.

Comment (juliogonzalez) [reply lib/gitarro/backend.rb]: Ah! Right.

Let's address that with a separate PR. I think there's room for a lot improvement there, not only whitespaces and case sensitivity.

**Comment by juliogonzalez** [comment]
Ok, let's merge. I will not release this until #169 is merged.

**Pull request merged**